packages/smooth_app/lib/pages/onboarding/onboarding_flow_navigator.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'package:smooth_app/data_models/user_preferences.dart';
+import 'package:smooth_app/pages/onboarding/scan_example.dart';
+import 'package:smooth_app/pages/onboarding/welcome_page.dart';
+import 'package:smooth_app/pages/page_manager.dart';
+
+enum OnboardingPage {
+  NOT_STARTED,
+  WELCOME,
+  SCAN_EXAMPLE,
+  ONBOARDING_COMPLETE,
+}
+
+/// Decide which page to take the user to.
+class OnboardingFlowNavigator {
+  OnboardingFlowNavigator(this._userPreferences);
+
+  final Future<UserPreferences> _userPreferences;
+
+  Widget getNextPageWidget(BuildContext context, OnboardingPage currentPage) {
```

Comment:
It's not the "next" page, it's the page you're asking.
`getPageWidget` would make more sense.

Reply:
Actually it is the `next` page, we supply the `currentPage` to the function and it gets us the widget for `nextPage`